Source/Applications/SystemCenter/Controllers/OpenXDA/Assets/OpenXDAAssetController.cs

```diff
@@ -622,7 +622,35 @@ public IHttpActionResult PostExistingLocationForAsset(int assetID, int locationI
             }
         }
 
+        [HttpGet, Route("{assetID:int}/ConnectedChannels")]
+        public IHttpActionResult GetAssetChannels(int assetID)
+        {
+            if (GetRoles == string.Empty || User.IsInRole(GetRoles))
+            {
+
+                using (AdoDataConnection connection = new AdoDataConnection(Connection))
+                {
+                    Asset asset = new TableOperations<Asset>(connection).QueryRecordWhere("ID={0}", assetID);
+                    if (asset is null)
+                        return Ok();
```

Comment:
@gcsantos-gpa in this case We should probably throw an exception saying the Asset does not exist instead